google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryAggregationsTest.java

```diff
@@ -1,5 +1,750 @@
+import static java.util.Arrays.asList;
 
+import com.google.cloud.firestore.*;
+import com.google.common.base.Preconditions;
+import java.util.Map;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class ITQueryAggregationsTest {
+  private Firestore firestore;
+
+  @Before
+  public void setUpFirestore() {
+    // TODO: stop using emulator for these tests once prod is ready.
+    firestore = FirestoreOptions.newBuilder().setHost("localhost:8080").build().getService();
+    Preconditions.checkNotNull(
+        firestore,
+        "Error instantiating Firestore. Check that the service account credentials were properly set.");
+  }
+
+  @After
+  public void tearDownFirestore() throws Exception {
+    if (firestore != null) {
+      firestore.close();
+      firestore = null;
+    }
+  }
+
+  private CollectionReference testCollection() {
+    String collectionPath = "java-" + autoId();
+    return firestore.collection(collectionPath);
+  }
+
+  private CollectionReference testCollection(String name) {
+    return firestore.collection("java-" + name + "-" + autoId());
+  }
+
+  private CollectionReference testCollectionWithDocs(Map<String, Map<String, Object>> docs)
+      throws InterruptedException {
+    CollectionReference collection = testCollection();
+    CollectionReference writer = firestore.collection(collection.getId());
+    writeAllDocs(writer, docs);
+    return collection;
+  }
+
+  public static void writeAllDocs(
+      CollectionReference collection, Map<String, Map<String, Object>> docs)
+      throws InterruptedException {
+    for (Map.Entry<String, Map<String, Object>> doc : docs.entrySet()) {
+      await(collection.document(doc.getKey()).set(doc.getValue()));
+    }
+  }
+
+  private static Map<String, Map<String, Object>> testDocs1 =
+      map(
+          "a",
+          map("author", "authorA", "title", "titleA", "pages", 100, "foo", 1, "bar", 2, "baz", 3),
+          "b",
+          map("author", "authorB", "title", "titleB", "pages", 50, "foo", 1, "bar", 2, "baz", 3));
+
+  @Test
+  public void canRunCountUsingAggregationMethod() throws Exception {
+    CollectionReference collection = testCollectionWithDocs(testDocs1);
+    AggregateQuerySnapshot snapshot = collection.aggregate(AggregateField.count()).get().get();
+    assertThat(snapshot.getCount()).isEqualTo(2);
+  }
+
+  @Test
+  public void canGetDuplicateAggregations() throws Exception {
+    CollectionReference collection = testCollectionWithDocs(testDocs1);
+    AggregateQuerySnapshot snapshot =
+        collection
+            .aggregate(AggregateField.count(), AggregateField.count(), sum("pages"), sum("pages"))
+            .get()
+            .get();
+    assertThat(snapshot.getCount()).isEqualTo(2);
+    assertThat(snapshot.get(sum("pages"))).isEqualTo(150);
+  }
+
```

Comment:
We should probably port the test ['getAggregateFromServer error message is good if missing index'](https://github.com/firebase/firebase-js-sdk/pull/6952/files?diff=unified&w=0#diff-a9bd0f4b702f0f7af954b8b1a7db64bd9faf9e521c1f1719a73c9d79effc5fd7R310)